pkg/local_workflows/filter_workflow.go

```diff
@@ -0,0 +1,152 @@
+package localworkflows
+
+import (
+	"encoding/json"
+	"slices"
+	"strings"
+
+	"github.com/snyk/error-catalog-golang-public/snyk_errors"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/content_type"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/local_models"
+	"github.com/snyk/go-application-framework/pkg/utils"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/pflag"
+)
+
+const (
+	FilterFindingsWorkflowName = "findings.filter"
+)
+
+var WORKFLOWID_FILTER_FINDINGS = workflow.NewWorkflowIdentifier(FilterFindingsWorkflowName)
+
+func InitFilterFindingsWorkflow(engine workflow.Engine) error {
+	flags := pflag.NewFlagSet(FilterFindingsWorkflowName, pflag.ExitOnError)
+	_, err := engine.Register(WORKFLOWID_FILTER_FINDINGS, workflow.ConfigurationOptionsFromFlagset(flags), filterFindingsEntryPoint)
+
+	return err
+}
+
+type FindingsFilterFunc func(local_models.FindingResource) bool
+
+func filterSeverityASC(original []string, severityMinLevel string) []string {
+	if severityMinLevel == "" {
+		return original
+	}
+
+	minLevelPointer := slices.Index(original, severityMinLevel)
+
+	if minLevelPointer >= 0 {
+		return original[minLevelPointer:]
+	}
+
+	return original
+}
+
+func getSeverityThresholdFilter(severityThreshold string, severityOrder []string) FindingsFilterFunc {
+	return func(finding local_models.FindingResource) bool {
+		allowed_severities := filterSeverityASC(severityOrder, severityThreshold)
+
+		return utils.Contains(allowed_severities, string(finding.Attributes.Rating.Severity.Value))
+	}
+}
+
+func applyFilters(findingsModel *local_models.LocalFinding, filters []FindingsFilterFunc) {
+	// filteredFindings := local_models.LocalFinding{}
+	filteredFindings := []local_models.FindingResource{}
```

Comment:
Issue: I'm just realizing that we need to filter the summary as well, not just only the findings. 
This means, the summary contains ignored and open issues by severity, if we apply a severity filter, it needs to adapt both (findings and summary). 

For the sarif based flow, this is already done in the [output workflow](https://github.com/snyk/go-application-framework/blob/main/pkg/local_workflows/output_workflow.go#L50).

Reply:
I think this is related to the bug ticket CLI-559 that I created.

Reply:
I think the solution is relatively easy, as it would just require to change the filter interface a bit to accept the complete findings model. Iterating through the different data fields can than be done in the filter.
Positive side effect, this filter interface is even more powerful and suitable for other filter types as well.

Reply:
we'll do this as a separate PR as part of CLI-559